### ISSUE TITLE
`addRequiredGroup` not working with nested usage

### ISSUE THREAD

**Issue by Asim-Tahir** [horprogs/Just-validate#52]
**Describe the bug**
`addRequiredGroup` method throws an error when nesting with checkbox required group, even if the required group contains checkbox(es) or radio(s).

**To Reproduce**
1. Go to https://codepen.io/AsimTahir/pen/vYWVOLB
2. Open the Browser's Console tab
3. See error
    ```log
    just-validate.production.min.js:1 Uncaught Error: Group should contain either or checkboxes or radio buttons
        at O.addRequiredGroup (just-validate.production.min.js:1:17980)
        at pen.js:9:12
    ```

**Expected behavior**
Both nested required groups should be initialized properly.

**Additional context**
##### Error trackback
1. The error came cause by `isRadio` and `isCheckbox` variables.
https://github.com/horprogs/Just-validate/blob/de6039b16b8bccc32093e2688387d9b629464c5d/src/main.ts#L1080-L1086
The `inputs` variable contains checkbox or radio input but both `isRadio` and `isCheckbox` variables return `false`.

**Temporary solution**
```ts
import JustValidate from "just-validate";

class JustValidateExtended extends JustValidate {
  addRequiredGroup(
    groupField: string,
    errorMessage?: string,
    config?: FieldConfigInterface,
    successMessage?: string
  ): JustValidate {
        if (typeof groupField !== "string") {
            throw Error(
                `Group selector is not valid. Please specify a string selector.`
            );
        }

        const elem = this.form!.querySelector(groupField) as HTMLElement;

        if (!elem) {
            throw Error(
                `Group with ${groupField} selector not found! Please check the group selector.`
            );
        }

        const inputs = elem.querySelectorAll("input[type=radio],input[type=checkbox]");

        const isRadio = Array.from(inputs).some((input) => input.type === 'radio');
        const isCheckbox = Array.from(inputs).some((input) => input.type === 'checkbox');
        // or
        // const isRadio = elem.querySelectorAll("input[type=radio]").length > 0;
        // const isCheckbox = elem.querySelectorAll("input[type=checkbox]").length > 0;

        if (!isRadio && !isCheckbox) {
            throw Error(
                `Group should contain either or checkboxes or radio buttons`
            );
        }

        this.groupFields[groupField] = {
            rules: [
                {
                    rule: "required",
                    errorMessage,
                    successMessage,
                },
            ],
            groupElem: elem,
            elems: Array.from(inputs),
            type: isRadio ? "radio" : "checkbox",
            isDirty: false,
            isValid: undefined,
            config,
        };

        inputs.forEach((input) => {
            this.setListeners(input);
        });

        return this;
    }
}

window.JustValidate = JustValidateExtended;
```

**Comment by horprogs** [comment]
Thank you for your PR and sorry for the delay. I will look into this a bit later, as I feel it could be some complications

**Comment by Asim-Tahir** [comment]
Any update on this.

**Comment by horprogs** [comment]
Working on this. I implemented some solution, but it's not ideal. This is a bit tricky thing to have nested groups, because we cannot really exclude most nested group inputs from being in the parent group. That's why there are some edge cases when we don't show errors correctly, for example. I'll try to find better solution.

**Comment by horprogs** [comment]
Released the fix `3.6.2`. Seems work fine.

Also I checked your case in particular and it was a bit specific about the framework `Alpine.js` which you use.
 
First, to make it work you should use `x-if`, instead of `x-show`, because the inputs shouldn't be in DOM at all, otherwise they will belong to the wrong group. 

And second, I found a weird behaviour, because `Alpine.js` dispatches a change event (and triggers addRequiredGroup method) before the actual click input event fires. I have managed to overcome this using `setTimeout` for `addRequiredGroup`, so it for sure happens after the click input event fires.

You could check how it works with your case here:
https://codepen.io/Horprogs/pen/yLvYMGV

**Comment by Asim-Tahir** [comment]
Thx, it seems working. I test it ASAP. If the problem still occurs I will reopen a new issue 👍
